### PR TITLE
(Finally Ready for Review) __init__ file support and import *

### PR DESCRIPTION
**Pull request by KevinHock**

### CODE REVIEW

File: pyt/utils/log.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import logging
```
Comment (KevinHock) [inline pyt/utils/log.py]: This file is completely unused but while developing you can do

```python
from pyt.utils.log import enable_logger, logger
enable_logger(to_file='./pyt.log')
```

to log to a file.

File: pyt/cfg.py [hunk 2]
```diff
@@ -1,64 +0,0 @@
-"""This module contains classes for generating a Control Flow Graph
```
Comment (KevinHock) [inline pyt/cfg.py]: Dead file.

File: example/import_test_project/init_3_with_alias.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import init_file_folder_3_with_alias
```
Comment (KevinHock) [inline example/import_test_project/init_3_with_alias.py]: This is the one that doesn't pass.

Comment (KevinHock) [inline example/import_test_project/init_3_with_alias.py]: Yet.

File: pyt/interprocedural_cfg.py [hunk 4]
```diff
@@ -412,30 +420,124 @@ def add_file_module(self, module, module_name, local_names, import_alias_mapping
...
+        # Remember, module[0] is None during e.g. "from . import foo", so we must str()
+        self.append_node(EntryOrExitNode('Module Entry ' + str(module[0])))
         self.visit(tree)
-        exit_node = self.append_node(EntryOrExitNode('Module Exit ' + module[0]))
+        exit_node = self.append_node(EntryOrExitNode('Module Exit ' + str(module[0])))
 
         self.module_definitions_stack.pop()
         self.filenames.pop()
 
+        if module_definitions.is_init:
```
Comment (KevinHock) [inline pyt/interprocedural_cfg.py]: I'm gonna clean this up a bit, don't worry.

File: pyt/interprocedural_cfg.py [hunk 5]
```diff
@@ -444,86 +546,71 @@ def visit_Import(self, node):
             for module in self.local_modules:
                 if name.name == module[0]:
                     if os.path.isdir(module[1]):
-                        return self.import_directory_module(module,
```
Comment (KevinHock) [inline pyt/interprocedural_cfg.py]: Which one is better? with or without the else?

**Comment by KevinHock** [comment]
Yay, just got the last test passing, gonna clean it up and push tomorrow, the problem was not passing the folder name [here](https://github.com/python-security/pyt/pull/39/files#diff-50f071a5d7996bb4ad86dac436550d9aR507). :D

File: pyt/interprocedural_cfg.py (hunk 5 above)
Comment (KevinHock) [inline pyt/interprocedural_cfg.py]: Coworkers and pylint think without, so without.

**Comment by KevinHock** [comment]
Gonna clean up that last commit, but before I do, I might as well support `from foo import *`, I don't care about `_all_` and the fact that names with a leading underscore aren't imported (because if code calls something that's not in _all_ or has a leading underscore there would be an error anyway, right?) so I'm not going to support those at this point in time.

**Comment by Toughee** [comment]
Okay. Thanks for the advice @KevinHock

File: example/import_test_project/init_3_with_alias.py (hunk 3 above)
Comment (KevinHock) [inline example/import_test_project/init_3_with_alias.py]: (passes now ;)

**Comment by Thalmann** [comment]
Looks great, and some awesome tests :)

I do not like the names of the different init_file_folder_1, init_file_folder_2 etc.
What about calling it what it is? For instance: init_file_folder, init_file_folder_call_module_attribute, etc?

**Comment by KevinHock** [comment]
Thanks, I can rename the tests, there's also the 2 functions in 'alias_helper.py' that do the same thing except the keys and values are switched around in their 'import_alias_mapping's, so I just need to switch one and then can have 1 function instead of 2. 

Any other cleanups?

**Comment by KevinHock** [comment]
So the 2 `import_alias_mapping` function helpers aren't as easy to merge as I thought, anyways, the renaming has occurred :)

**Pull request merged**